File: packages/sdk/src/administrator/roles.ts
import { QelosSDKOptions } from '../types';
import BaseSDK from '../base-sdk';

export class QlRoles extends BaseSDK {
  private relativePath = '/api/roles';

  constructor(private options: QelosSDKOptions) {
    super(options);
  }

 // Method to get all existing roles
  getExistingRoles() {
    return this.callJsonApi<string[]>(this.relativePath);
  }
}

File: packages/python-sdk/qelos_sdk/administrator/roles.py
from __future__ import annotations

from typing import List

from ..base_sdk import BaseSDK
from ..types import QelosSDKOptions


class QlRoles(BaseSDK):
    """Admin role management."""

    _relative_path = "/api/roles"

    def __init__(self, options: QelosSDKOptions) -> None:
        super().__init__(options)

    async def get_existing_roles(self) -> List[str]:
        """Get all existing roles."""
        return await self.call_json_api(self._relative_path)
